### PR TITLE
add "humility sbrmi" + "humility dashboard" fixes

### PR DESCRIPTION
**Pull request by bcantrill**
Draft comment:

```
In addition to adding the new "humility sbrmi" command, this work
fixes some minor "humility dashboard" issues when logging to an output
file (`--output` option): it adds a timestamp, power state, and
displays all sensors in the CSV header (not merely temp sensors).

Finally, this pulls in the fix for probe-rs/probe-rs#1517, which
allows for ST parts that have writeback caches (e.g., the H7 family)
to be correctly used with non-STLink probes (e.g., CMSIS-DAP probes
like the MCU Link).
```

### CODE REVIEW

File: cmd/sbrmi/src/cpuid.rs [hunk 1]
```diff
@@ -0,0 +1,1350 @@
+//
+// The MIT License (MIT)
+//
+// Copyright (c) 2015 Gerd Zellweger
```
Comment (mkeeter) [inline cmd/sbrmi/src/cpuid.rs:4]: https://github.com/gz/rust-cpuid/pull/137 is the upstream PR that puts formatting into the upstream crate.

File: cmd/sbrmi/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+//!                          Hardware ID                         = 0xb0
+//!                          Instance ID (low)                   = 0x0
+//!
+//!     MCA_DESTAT(0)  0x0000000000000000
+//!     MCA_DEADDR(0)  0x0000000000000000
+//! ```
+//!
+//! CPU identification information as provided by the `cpuid` instruction
+//! can be retrieved by using the `--cpuid` option and specifying a
+//! desired target thread; full MCA inbformation can similarly be retrieved
```
Comment (mkeeter) [inline cmd/sbrmi/src/lib.rs:89]: Typo:
```suggestion
//! desired target thread; full MCA information can similarly be retrieved
```
(requires rebuilding the README as well)

File: cmd/sbrmi/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+    core: &mut dyn Core,
+    context: &mut HiffyContext,
+    thread: Option<u8>,
+) -> Result<()> {
+    let thread = thread.unwrap_or(0);
+
+    //
+    // We want to call into the raw_cpuid crate to pull and display CPUID
+    // information, but it unfortunately takes only a function, not a
+    // closure.  (It -- understndably -- didn't forsee the CPUID instruction
```
Comment (mkeeter) [inline cmd/sbrmi/src/lib.rs:198]:
```suggestion
    // closure.  (It -- understandably -- didn't forsee the CPUID instruction
```

File: cmd/sbrmi/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,686 @@
...
+    // being run via a sidechannel I2C interface through an intermediary
+    // operating at a distance!)  We resort to the same technique used by the
+    // Hiffy RPC engine for its client-side HIF emulation:  we create a
+    // structure to hold the pointers to things that we need, and fill it with
+    // more or less raw pointers, transmuting away all lifetimes in the
+    // process.  We dispense with some of the precaution taken in that code
+    // because this isn't callable by anyone else, and the stakes here are
+    // generally low: we're going to call into the raw_cpuid code to
+    // prett-print the results of the proxied CPUID, and then we're going to
```
Comment (mkeeter) [inline cmd/sbrmi/src/lib.rs:207]:
```suggestion
    // pretty-print the results of the proxied CPUID, and then we're going to
```

File: cmd/sbrmi/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,686 @@
...
+                    context: None,
+                    thread: None,
+                });
+    }
+
+    fn cb(eax: u32, ecx: u32) -> CpuIdResult {
+        SBRMI_WORKSPACE.with(|workspace| {
+            let workspace = workspace.borrow_mut();
+            let (hubris, core, context) = {
+                // Hold Matt's beer
```
Comment (mkeeter) [inline cmd/sbrmi/src/lib.rs:232]: 🍺

**Review by mkeeter — APPROVED** [review]
LGTM, other than a few typos

File: cmd/sbrmi/src/lib.rs (hunk 2 above)
Comment (bcantrill) [reply cmd/sbrmi/src/lib.rs:89]: Done

File: cmd/sbrmi/src/lib.rs (hunk 4 above)
Comment (bcantrill) [reply cmd/sbrmi/src/lib.rs:207]: Done!

File: cmd/sbrmi/src/lib.rs (hunk 5 above)
Comment (bcantrill) [reply cmd/sbrmi/src/lib.rs:232]: 🍻!

File: cmd/sbrmi/src/lib.rs (hunk 3 above)
Comment (bcantrill) [reply cmd/sbrmi/src/lib.rs:198]: Done -- thank you!

**Pull request merged**